test/CMakeLists.txt

```diff
@@ -1,100 +1,46 @@
 # SPDX-License-Identifier: Apache-2.0
-
-#####################################################
-#################### GETEST #######################
-#####################################################
-enable_testing()
-if(KOMPUTE_OPT_REPO_SUBMODULE_BUILD)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/external/googletest EXCLUDE_FROM_ALL
-        ${CMAKE_CURRENT_BINARY_DIR}/kompute_googletest)
-else()
-    find_package(GTest CONFIG REQUIRED)
-endif()
-
-file(GLOB test_kompute_CPP
-    "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp"
-)
-
-add_executable(test_kompute ${test_kompute_CPP})
-
-target_include_directories(
-    test_kompute PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/single_include>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/compiled_shaders_include>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/utils>
-)
-
-if(KOMPUTE_OPT_REPO_SUBMODULE_BUILD)
-    target_include_directories(
-        test_kompute PRIVATE
-        ${gtest_SOURCE_DIR}/include)
-
-    target_link_libraries(test_kompute
-        gtest_main)
-else()
-    target_link_libraries(test_kompute
-        GTest::gtest)
-endif()
-
-target_link_libraries(test_kompute kompute)
-
-add_test(NAME test_kompute COMMAND test_kompute)
-
-
-#####################################################
-#################### CODECOV  #######################
-#####################################################
-
-if (KOMPUTE_OPT_CODE_COVERAGE)
-    if(NOT UNIX)
-        message(
-            FATAL_ERROR
-            "KOMPUTE_OPT_CODE_COVERAGE can only be enabled in unix based systems due to limitation on gcov")
+# ######################
```

Comment:
I am currently running into some errors when building - when I build using 

```
cmake --build build/ --target all
```

I get the error:

```
&, const long unsigned int&, const long unsigned int&, const long unsigned int&}]’
/home/.../kompute/build/_deps/vulkan_header-src/include/vulkan/vulkan_structs.hpp:52917:44:   required from here
/usr/include/c++/7/type_traits:1663:12: fatal error: template instantiation depth exceeds maximum of 900 (use -ftemplate-depth= to increase the maximum)
     struct add_rvalue_reference
            ^~~~~~~~~~~~~~~~~~~~

```

Have you seen this before?

Reply:
Like discussed, this is a Vulkan version conflict. It can be solved by downgrading Vulkan headers used for compiling via
`cmake .. -DKOMPUTE_OPT_BUILT_IN_VULKAN_HEADER_TAG="vx.y.z"